api/rpc/stat/stat_test.go

```diff
@@ -1 +1,72 @@
 package stat
+
+import (
+	"net"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc"
+)
+
+const (
+	svcAddr  = "localhost:51001"
+	queryVal = "measurements"
+)
+
+// Create a standalone stat grpc server for test purposes
+func TestMain(m *testing.M) {
+	//check port
+	lis, err := net.Listen("tcp", svcAddr)
+	if err != nil {
+		panic(err)
+	}
+	//Create inline server
+	s := grpc.NewServer()
+	//Register Server
+	RegisterStatServer(s, createStatServer())
+	//Launch Server with defered stop
+	go s.Serve(lis)
```

Comment:
Not sure it's worth all the extra boilerplate just to explicitly start one service instead of all of them when the server starts as fast as it does. You sure don't just want to start the server like [this test](https://github.com/appcelerator/amp/blob/master/data/storage/etcd/store_test.go#L53)? Maybe we ought to just create a utility function for testing that just starts a single rpc service on a port that at least doesn't require copying all this boilerplate over for each test. And frankly, until we're doing actual integration tests that goes through our client code to talk to a running server and actually test our network interface end-to-end, we really don't even need to be starting a server at all right now.